metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -2299,6 +2364,11 @@ def emit_published_datasources(self) -> Iterable[MetadataWorkUnit]:
             c.PUBLISHED_DATA_SOURCES_CONNECTION,
             datasource_filter,
         ):
+            datasource = self.update_datasource_for_field_upstream(
```

Comment:
how many additional gql queries do we send to tableau? how much does that impact our overall ingestion times?

Reply:
Lets consider if we have n datasources and each datasources has m fields then previously we were making `n/page_size` API calls and now it would be `(n + n*m)/page_size`.

For example if n = 100 and each datasource has 20 fields and page_size is 10
     OLD Api Count = 100/10 = 10 Api Call 
     New Api Count = (100 + 100 * 20 ) / 10 = 210

We can control the api calls using page_size. Over ingestion time is depend on multiple factors (tableau node configuration + ingestion node configuration) and 10 Api call with more memory utilization vs 210 Api call with less memory utilization.

if you need exact number I can generate the figures using our tableau instance and running ingestion on my machine. Let me know if we want to run this exercise ?